src/odemis/gui/win/acquisition.py

```diff
@@ -612,6 +612,409 @@ def on_acquisition_done(self, future):
+        Must be called in the main GUI thread.
+        """
+        # stop listening to events
+        self._view.stream_tree.flat.unsubscribe(self.on_streams_changed)
+        self._hidden_view.stream_tree.flat.unsubscribe(self.on_streams_changed)
+        self.stop_listening_to_va()
+
+        self.remove_all_streams()
+
+    def on_close(self, evt):
+        """ Close event handler that executes various cleanup actions
+        """
+        if self.acq_future:
+            # TODO: ask for confirmation before cancelling?
+            # What to do if the acquisition is done while asking for
+            # confirmation?
+            msg = "Cancelling acquisition due to closing the acquisition window"
+            logging.info(msg)
+            self.acq_future.cancel()
+
+        self.terminate_listeners()
+
+        self.EndModal(wx.ID_CANCEL)
+
+    def _pause_settings(self):
+        """ Pause the settings of the GUI and save the values for restoring them later """
+        self._settings_controller.pause()
+        self._settings_controller.enable(False)
+
+        self.streambar_controller.pause()
+        self.streambar_controller.enable(False)
+
+    def _resume_settings(self):
+        """ Resume the settings of the GUI and save the values for restoring them later """
+        self._settings_controller.enable(True)
+        self._settings_controller.resume()
+
+        self.streambar_controller.enable(True)
+        self.streambar_controller.resume()
+
+    def on_acquire(self, evt):
+        """ Start the actual acquisition """
+        if self.data:  # This means the button is actually "View"
+            self._view_data()()
+            return
+
+        logging.info("Acquire button clicked, starting acquisition")
+        self.acquiring = True
+
+        self.btn_secom_acquire.Disable()
+
+        # disable estimation time updates during acquisition
+        self._view.lastUpdate.unsubscribe(self.on_streams_changed)
+
+        # Freeze all the settings so that it's not possible to change anything
+        self._pause_settings()
+
+        self.gauge_acq.Show()
+        self.Layout()  # to put the gauge at the right place
+
+        # For now, always indicate the best quality (even if the preset is set
+        # to "live")
+        if self._main_data_model.opm:
+            self._main_data_model.opm.setAcqQuality(path.ACQ_QUALITY_BEST)
+
+        # Note: It should never be possible to reach here with no streams
+        streams = self.get_acq_streams()
+        v_streams = self._view.getStreams()  # visible streams
+        for s in streams:
+            # Add extra viewable streams to view. However, do not add incompatible streams.
+            if s not in v_streams and not isinstance(s, NON_SPATIAL_STREAMS):
+                self._view.addStream(s)
+
+        # TODO: Test with merged code from stitching
+        self.acq_future, est_time, est_memory = stitching.acquireOverviewMap(streams, self._main_data_model.stage)
```

Comment:
Check with Bassim, but I think that EM and FM acquisitions need to be done apart. So that's a little more complex, both here and for the progress bar.